streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -177,63 +177,96 @@ public KStreamImpl(KStreamBuilder topology, String name, Set<String> sourceNodes
 
     @Override
     public void print() {
-        print(null, null, null);
+        print(null, null, null, null);
     }
 
     @Override
     public void print(String streamName) {
-        print(null, null, streamName);
+        print(null, null, null, streamName);
     }
 
     @Override
     public void print(Serde<K> keySerde, Serde<V> valSerde) {
-        print(keySerde, valSerde, null);
+        print(null, keySerde, valSerde, null);
     }
 
     @Override
     public void print(Serde<K> keySerde, Serde<V> valSerde, String streamName) {
+        print(null, keySerde, valSerde, streamName);
+    }
+
+    @Override
+    public void print(KeyValueMapper<? super K, ? super V, String> mapper) {
+        print(mapper, null, null, null);
+    }
+
+    @Override
+    public void print(KeyValueMapper<? super K, ? super V, String> mapper, String streamName) {
+        print(mapper, null, null, streamName);
+    }
+
+    @Override
+    public void print(KeyValueMapper<? super K, ? super V, String> mapper, Serde<K> keySerde, Serde<V> valSerde) {
+        print(mapper, keySerde, valSerde, null);
+    }
+
+    @Override
+    public void print(KeyValueMapper<? super K, ? super V, String> mapper, Serde<K> keySerde, Serde<V> valSerde, String streamName) {
         String name = topology.newName(PRINTING_NAME);
         streamName = (streamName == null) ? this.name : streamName;
-        topology.addProcessor(name, new KStreamPrint<>(new PrintForeachAction(null, streamName), keySerde, valSerde), this.name);
+        topology.addProcessor(name, new KStreamPrint<>(new PrintForeachAction(null, mapper, streamName), keySerde, valSerde), this.name);
     }
 
-
     @Override
     public void writeAsText(String filePath) {
-        writeAsText(filePath, null, null, null);
+        writeAsText(filePath, null, null, null, null);
     }
 
     @Override
     public void writeAsText(String filePath, String streamName) {
-        writeAsText(filePath, streamName, null, null);
+        writeAsText(filePath, streamName, null, null, null);
```

Comment:
The reason is if I pass in KeyValueMapper instead of passing null, I should create a instance in `print()` and `writeAsText()` which is 
```
final KeyValueMapper<K, V, String> mapper = new KeyValueMapper<K, V, String>() {
          @Override
          public String apply(K key, V value) {
              return String.format("%s, %s", key, value);
          }
 };
```
and I think it will make code unclean.